R/create_base_vectors.R

```diff
@@ -6,294 +6,416 @@
-#' #  } else {
-#' #    print("No transmission lines in area of interest")
-#' #  }
-# }
+#' create_base_vectors(sn_aoi, out_dir = PEMprepr::read_fid()$dir_1010_vector$path_abs)
+#'}
+create_base_vectors <- function(aoi = fs::path(PEMprepr::read_fid()$dir_1010_vector$path_abs, "aoi_snapped.gpkg"),
+                                out_dir = PEMprepr::read_fid()$dir_1010_vector$path_abs) {
+  # # testing
+  # aoi <- fs::path(PEMprepr::read_fid()$dir_1010_vector$path_abs, "aoi_snapped.gpkg")
+  #  aoi <- sf::st_read(aoi, quiet = TRUE)
+  #  out_dir <- PEMprepr::read_fid()$dir_1010_vector$path_abs
+
+  if (missing(aoi)) stop("'aoi' is missing with no default")
+
+  # Second, detect object type and convert where necessary
+  if (!inherits(aoi, c("sf", "sfc"))) {
+    stop("'aoi' is not an sf or sfc object.")
+  }
+  if (is.null(out_dir)) {
+    stop("\rout_dir is an invalid file path string")
+  }
+  # Detect the CRS of the sf object
+  if (is.na(sf::st_crs(aoi))) {
+    stop("CRS is not assigned. Use sf::st_crs() to assign a valid CRS to aoi")
+  }
+  if (sf::st_is_longlat(aoi)) {
+    cat("Input CRS is Lat/Long format. Transforming to EPSG 3005 (BC Albers) for processing\n")
+    epsg <- 3005L
+    in_crs <- sf::st_crs(aoi)
+    aoi <- sf::st_transform(aoi, epsg) %>% sf::st_set_agr("constant")
+  } else {
+    in_crs <- sf::st_crs(aoi)
+    epsg <- in_crs$epsg
+    aoi <- sf::st_set_agr(aoi, "constant")
+    if (!is.numeric(epsg)) {
+      stop("There was a problem retrieving the EPSG code from the aoi. Is it assigned properly?")
+    }
+  }
+
+  epsg <- sf::st_crs(aoi)
+
+  # note this was an earlier flag - I dont like the idea of changing internals
+  # @Andy T. any thoughts on speed improvements.
+
+  # Adjust max download size based on AOI
+  ## PROBLEMATIC -- should not be done globally -----------------
+  ## see: https://r-pkgs.org/code.html#sec-code-r-landscape section 7.6.1
+  # withr::local_options(
+  #   options(bcdata.max_geom_pred_size = as.numeric(st_area(aoi)) + 10)
+  # )
+
+  get_BEC(aoi, out_dir)
+  get_VRI(aoi, out_dir)
+  get_harvest(aoi, out_dir)
+  get_TEM(aoi, out_dir)
+  get_water(aoi, out_dir)
+  get_roads(aoi, out_dir)
+  get_towns(aoi, out_dir)
+  get_fires(aoi, out_dir)
+  get_fire_severity(aoi, out_dir)
+  get_parks(aoi, out_dir)
+  get_transmission_lines(aoi, out_dir)
+
+  cli::cat_line()
+  cli::cli_alert_success(
+    "Layers downloaded and to written to {.path {out_dir}}"
+  )
+}
+
+### 1) Get_BEC ----------------------------
+get_BEC <- function(aoi, out_dir) {
+  #message("\rDownloading BEC layers")
+
+  # # 1) BEC Biogeographical linework
+  bec <- bcdata::bcdc_query_geodata("f358a53b-ffde-4830-a325-a5a03ff672c3") |>
+    bcdata::filter(bcdata::INTERSECTS(aoi)) |>
+    bcdata::select("MAP_LABEL") |>
+    bcdata::collect() |>
+    dplyr::select("MAP_LABEL") %>%
+    {
+      if (nrow(.) > 0) sf::st_intersection(., aoi) else .
+    }
```

Comment:
```suggestion
    dplyr::select("MAP_LABEL")
  
  if (nrow(bec) > 0) {
    bec <- sf::st_intersection(bec, aoi)
  }
```

Simplify: take it out of pipeline so avoid confusing `.` usage. What should happen when nrow == 0? The user should be alerted?

Reply:
This pattern applies to `get_TEM()`, `get_VRI()`, `get_fires()`, `get_harvest()`, `get_parks()`, `get_roads()`, `get_transmission_lines()`

Reply:
updates these to same format. I think adding a flag is a good next step where n ==0. I will add these in stage 2.